### ISSUE TITLE
stft gradient can produce nan

### ISSUE THREAD

**Issue by jokona** [KinWaiCheuk/nnAudio#138]
In nnAudio.features.stft, in STFT class, when output_format == "Magnitude", but trainable==False, the gradients will produce nan where spec == 0 in the last line of the following code:

```
if output_format == "Magnitude":
            spec = spec_real.pow(2) + spec_imag.pow(2)
            if self.trainable == True:
                return torch.sqrt(
                    spec + 1e-8
                )  # prevent Nan gradient when sqrt(0) due to output=0
            else:
                return torch.sqrt(spec)
```

this could be prevented by returning torch.sqrt(spec + 1e-8) like when trainable == False.

Is this behaviour intentional?

**Comment by migperfer** [comment]
Hi!

If trainable is False for the STFT, there should not be any gradients created, as the [kernels are created](https://github.com/KinWaiCheuk/nnAudio/blob/7ff8e82256ef9ffe203fb5274727f120ddfb1f74/Installation/nnAudio/features/stft.py#L234) with [torch.tensor](https://docs.pytorch.org/docs/stable/generated/torch.tensor.html#torch.tensor) which by default has requires_grad=False. 

Could you maybe share with us some code to reproduce the issue?

**Comment by jokona** [comment]
I no longer have access to the code that produced the error. Provided trainable=False produces no gradients, perhaps it was caused by a different issue. Thanks for the reply.
